modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartResourceImpl.java

```diff
@@ -257,6 +257,61 @@ private long _getCommerceRegionId(
 		return commerceRegion.getCommerceRegionId();
 	}
 
+	private CartItem[] _getValidatedCommerceOrderItems(
+			CommerceOrder commerceOrder)
+		throws Exception {
+
+		Map<Long, List<CommerceOrderValidatorResult>>
+			commerceOrderValidatorResults =
+				_commerceOrderValidatorRegistry.
+					getCommerceOrderValidatorResults(null, commerceOrder);
+
+		List<CommerceOrderItem> commerceOrderItems =
+			commerceOrder.getCommerceOrderItems();
+
+		List<CartItem> cartItems = new ArrayList<>();
+
+		for (CommerceOrderItem commerceOrderItem : commerceOrderItems) {
+			CartItem cartItem = _cartItemDTOConverter.toDTO(
+				new DefaultDTOConverterContext(
+					commerceOrderItem.getCommerceOrderItemId(),
+					contextAcceptLanguage.getPreferredLocale()));
+
+			if (commerceOrderValidatorResults.containsKey(
+					commerceOrderItem.getCommerceOrderItemId())) {
+
+				List<CommerceOrderValidatorResult>
+					commerceOrderItemValidatorResults =
+						commerceOrderValidatorResults.get(
+							commerceOrderItem.getCommerceOrderItemId());
+
+				Stream<CommerceOrderValidatorResult>
+					commerceOrderValidatorResultStream =
+						commerceOrderItemValidatorResults.stream();
+
+				cartItem.setValid(
+					commerceOrderValidatorResultStream.map(
+						commerceOrderItemValidatorResult ->
+							commerceOrderItemValidatorResult.isValid()
+					).reduce(
+						true, Boolean::logicalAnd
+					));
+				cartItem.setErrorMessages(
+					commerceOrderValidatorResultStream.map(
+						commerceOrderItemValidatorResult ->
+							commerceOrderItemValidatorResult.
+								getLocalizedMessage()
+					).toArray(
+						String[]::new
+					));
```

Comment:
New line here or sort alphabetical.
```suggestion
				cartItem.setValid(
					commerceOrderValidatorResultStream.map(
						commerceOrderItemValidatorResult ->
							commerceOrderItemValidatorResult.isValid()
					).reduce(
						true, Boolean::logicalAnd
					));

				cartItem.setErrorMessages(
					commerceOrderValidatorResultStream.map(
						commerceOrderItemValidatorResult ->
							commerceOrderItemValidatorResult.
								getLocalizedMessage()
					).toArray(
						String[]::new
					));
```